tests/query_runner/test_tinybird.py

```diff
@@ -0,0 +1,123 @@
+import json
+from unittest import TestCase
+from unittest.mock import Mock, patch
+
+from redash.query_runner import TYPE_DATETIME, TYPE_INTEGER, TYPE_STRING
+from redash.query_runner.tinybird import Tinybird
+
+DATASOURCES_RESPONSE = {
+    "datasources": [
+        {
+            "id": "t_datasource_id",
+            "name": "test_datasource",
+            "columns": [
+                {
+                    "name": "string_attribute",
+                    "type": "String",
+                    "nullable": False,
+                    "normalized_name": "string_attribute",
+                },
+                {
+                    "name": "number_attribute",
+                    "type": "Int32",
+                    "nullable": False,
+                    "normalized_name": "number_attribute",
+                },
+                {"name": "date_attribute", "type": "DateTime", "nullable": False, "normalized_name": "date_attribute"},
+            ],
+        }
+    ]
+}
+
+PIPES_RESPONSE = {"pipes": [{"id": "t_pipe_id", "name": "test_pipe", "endpoint": "t_endpoint_id", "type": "endpoint"}]}
+
+SCHEMA_RESPONSE = {
+    "meta": [
+        {"name": "string_attribute", "type": "String"},
+        {"name": "number_attribute", "type": "Int32"},
+        {"name": "date_attribute", "type": "DateTime"},
+    ]
+}
+
+QUERY_RESPONSE = {
+    **SCHEMA_RESPONSE,
+    "data": [
+        {"string_attribute": "hello world", "number_attribute": 123, "date_attribute": "2023-01-01 00:00:03.001000"},
+    ],
+    "rows": 1,
+    "statistics": {"elapsed": 0.011556914, "rows_read": 87919, "bytes_read": 17397219},
+}
+
+
+class TestTinybird(TestCase):
+    @patch("requests.get")
+    def test_get_schema(self, get_request):
```

Comment:
Both the test cases here are testing downstream functionality (i.e. `get_schema` and `run_query`, that are not directly implemented in the Tinybird runner). One potential improvement would be to test the Tinybird query runner directly, since it uses the base implementation for these methods.

Reply:
👍 I adjusted the test name to reflect what is happening in `get_schema` (it collects the schema from available pipes and datasources).

I am not sure what else to test when it comes to query execution though, as it's essentially just a tiny GET request which is already being tested for proper formatting and with mocked response handling 🤔